base/pkg/entry.jl

```diff
@@ -190,7 +188,7 @@ end
 
 function clone(url::AbstractString, pkg::AbstractString)
     info("Cloning $pkg from $url")
-    ispath(pkg) && throw(PkgError("$pkg already exists"))
+    ispath_casesensitive(pkg) && throw(PkgError("$pkg already exists"))
```

Comment:
We also want a check for `ispath(pkg)` here, because on a case-insensitive filesystem it will still get overwritten.